### PR TITLE
Adds go router push animations

### PR DESCRIPTION
**Pull request by chunhtai**
These assets will be used in go_router api doc

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/diagram_capture/lib/diagram_capture.dart [hunk 1]
```diff
@@ -75,47 +75,30 @@ class _Diagram extends StatelessWidget {
...
+const Size _kDefaultDiagramViewportSize = Size(1280.0, 1024.0);
 
 // View configuration that allows diagrams to not match the physical dimensions
 // of the device. This will change the view used to display the flutter surface
 // so that the diagram surface fits on the device, but it doesn't affect the
 // captured image pixels.
 class _DiagramViewConfiguration extends ViewConfiguration {
   _DiagramViewConfiguration({
-    super.constraints = _kDefaultDiagramViewportConstraints,
+    super.size = _kDefaultDiagramViewportSize,
```
Comment (chunhtai) [inline packages/diagram_capture/lib/diagram_capture.dart:86]: @gspencergoog I reverted some change you made, the dynamic view size pr is reverted and this change needs to be reverted too. I think we should make sure these tests are in flutter/test

Comment (gspencergoog) [reply packages/diagram_capture/lib/diagram_capture.dart:86]: Yeah, you're right, they should be in the customer tests.

**Comment by gspencergoog** [comment]
Are those animations generated by the diagram generator? If so, I don't see where they are generated. We generally want the diagrams and animations to be generated by code here so that we can just re-run them to get updated visuals. There are very few of the assets in this repo that aren't generated by code (mostly just input test photos, or native UI screen captures).

File: packages/diagram_capture/lib/diagram_capture.dart (hunk 1 above)
Comment (chunhtai) [reply packages/diagram_capture/lib/diagram_capture.dart:86]: I will write a pr to add to flutter/test

**Comment by chunhtai** [comment]
Hi @gspencergoog, yes they are generated from packages/diagrams/lib/src/go_router_push_animation.dart which is also part of this pr

**Comment by gspencergoog** [comment]
> Hi @gspencergoog, yes they are generated from packages/diagrams/lib/src/go_router_push_animation.dart which is also part of this pr

Oh, how embarrassing.  I just didn't see them in the PR (I neglected to scroll down far enough).

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**